src/components/shared/ui-components/UserForm.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useState,FormEvent } from "react";
+
+import { ReactNode } from "react";
+import "./UserForm.scss";
+import { useNavigate } from "react-router-dom";
+
+interface userFormProps {
+  fields: string[];
+  submitForm: (e: FormEvent<HTMLInputElement>) => void,
+  children: ReactNode,
+}
+const UserForm = ({ fields, submitForm ,children }: userFormProps) => {
+
```

Comment:
This is not generic enough to be a "shared" component.
a shared component usually is a ui component without logic